src/api/reports/attributes_report.py

```diff
@@ -0,0 +1,262 @@
+        "Biomass Constant C",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_family.name,
+                fish_family.biomass_constant_a,
+                fish_family.biomass_constant_b,
+                fish_family.biomass_constant_c,
+                *create_regions_row(regions, [str(r) for r in fish_family.regions]),
+            ]
+            for fish_family in FishFamily.objects.filter(status=SUPERUSER_APPROVED).order_by("name")
+        ],
+    ]
+
+    insert_range_and_resize(wb[FISH_FAMILY_NAME], "A1", data)
+
+
+def write_fish_genera(wb, regions):
+    region_names = list(regions.values())
+    COLUMN_NAMES = [
+        "Family",
+        "Name",
+        "Biomass Constant A",
+        "Biomass Constant B",
+        "Biomass Constant C",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_genus.family.name,
+                fish_genus.name,
+                fish_genus.biomass_constant_a,
+                fish_genus.biomass_constant_b,
+                fish_genus.biomass_constant_c,
+                *create_regions_row(regions, [str(r) for r in fish_genus.regions]),
+            ]
+            for fish_genus in FishGenus.objects.select_related("family")
+            .filter(status=SUPERUSER_APPROVED)
+            .order_by("family__name", "name")
+        ],
+    ]
+    insert_range_and_resize(wb[FISH_GENERA_NAME], "A1", data)
+
+
+def write_fish_species(wb, regions):
+    region_names = list(regions.values())
+    COLUMN_NAMES = [
+        "Family",
+        "Name",
+        "Biomass Constant A",
+        "Biomass Constant B",
+        "Biomass Constant C",
+        "Trophic Group",
+        "Functional Group",
+        "Max Length (cm)",
+        "Max Type",
+        "Group Size",
+        "Trophic Level",
+        "Vulnerability",
+        "Climate Score",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_species.genus.family.name,
+                fish_species.genus.name,
+                fish_species.name,
+                fish_species.biomass_constant_a,
+                fish_species.biomass_constant_b,
+                fish_species.biomass_constant_c,
+                fish_species.trophic_group and fish_species.trophic_group.name,
+                fish_species.functional_group and fish_species.functional_group.name,
+                fish_species.max_length,
+                fish_species.max_length_type,
+                fish_species.group_size and fish_species.group_size.name,
+                fish_species.trophic_level,
+                fish_species.vulnerability,
+                fish_species.climate_score,
+                *create_regions_row(
+                    regions, [str(r.id) for r in fish_species.regions.all()]
+                ),
+            ]
+            for fish_species in FishSpecies.objects.select_related(
```

Comment:
Compare queryset used in `FishSpeciesViewSet` -- seems like this should match (or even both use one definition?)

Reply:
I matched the quersets